src/Pact/Eval.hs

```diff
@@ -282,44 +282,57 @@ evaluateConstraints
   :: Info
   -> Module
   -> HM.HashMap Text Ref
-  -> Eval e ()
+  -> Eval e (Module, HM.HashMap Text Ref)
 evaluateConstraints info Interface{} _ =
   evalError info "Unexpected: interface found in module position while solving constraints"
-evaluateConstraints info Module{..} evalMap = foldMap evaluateConstraint _mInterfaces
+evaluateConstraints info m evalMap =
+  -- we would like the lazy semantics of foldr to shortcircuit the solver
+  foldr evaluateConstraint (pure (m, evalMap)) (_mInterfaces m)
   where
-    evaluateConstraint ifn = do
-      irefs <- preview $ eeRefStore . rsModules . ix ifn . mdRefMap
-      case irefs of
+    evaluateConstraint ifn em = do
+      (m',refMap) <- em
+      refData <- preview $ eeRefStore . rsModules . ix ifn
+      case refData of
         Nothing -> evalError info $
-          "Interface implemented in module, but not defined: <" ++ asString' ifn ++ ">"
-        Just irefs' -> HM.foldrWithKey (solveConstraint info evalMap) (pure ()) irefs'
+          "Interface implemented in module, but not defined: " ++ asString' ifn
+        Just (ModuleData Interface{..} irefs) -> do
+          em' <- HM.foldrWithKey (solveConstraint info) (pure refMap) irefs
+          let um = over mMeta (<> _interfaceMeta) m'
+          pure (um, em')
+        Just _ -> evalError info "Unexpected: module found in interface position while solving constraints"
 
 -- | Compare implemented member signatures with their definitions.
 -- At this stage, we have not merged consts, so we still check for overlap
 solveConstraint
   :: Info
-  -> HM.HashMap Text Ref
   -> Text
   -> Ref
-  -> Eval e ()
-  -> Eval e ()
-solveConstraint info _ refName (Direct t) _ =
+  -> Eval e (HM.HashMap Text Ref)
+  -> Eval e (HM.HashMap Text Ref)
+solveConstraint info refName (Direct t) _ =
   evalError info $ "found native reference " ++ show t ++ " while resolving module contraints: " ++ show refName
-solveConstraint info em refName (Ref t) _ =
+solveConstraint info refName (Ref t) evalMap = do
+  em <- evalMap
   case HM.lookup refName em of
-    Nothing -> pure ()
+    Nothing ->
+      case t of
+        TDef{..} ->
+          evalError info $ "found unimplemented member while resolving model constraints: " ++ show refName
+        _ -> evalMap
```

Comment:
@slpopejoy This was a bug I found while going through the logic of the `solveConstraint` step - in the case where reference names were not found, we were simply returning, but what we _should_ have been doing was checking to see if the reference `t` was a `TDef` and noting that it was an unimplemented member, and stepping over everything else.